paasta_tools/kubernetes_tools.py

```diff
@@ -2029,6 +2034,62 @@ def get_pod_management_policy(self) -> str:
         """Get sts pod_management_policy from config, default to 'OrderedReady'"""
         return self.config_dict.get("pod_management_policy", "OrderedReady")
 
+    def format_as_kubernetes_job(self) -> V1Job:
+        """Create the config for launching the deployment as a Job"""
+        try:
+            docker_url = self.get_docker_url()
+            git_sha = get_git_sha_from_dockerurl(docker_url, long=True)
+            system_paasta_config = load_system_paasta_config()
+            complete_config = V1Job(
+                api_version="batch/v1",
+                kind="Job",
+                metadata=self.get_kubernetes_metadata(git_sha),
+                spec=V1JobSpec(
+                    active_deadline_seconds=3600,
+                    selector=V1LabelSelector(
+                        match_labels={
+                            "paasta.yelp.com/service": self.get_service(),
+                            "paasta.yelp.com/instance": self.get_instance(),
+                        }
+                    ),
+                    template=self.get_pod_template_spec(
+                        git_sha=git_sha, system_paasta_config=system_paasta_config
+                    ),
+                ),
+            )
+
+            prometheus_shard = self.get_prometheus_shard()
+            if prometheus_shard:
+                complete_config.metadata.labels[
+                    "paasta.yelp.com/prometheus_shard"
+                ] = prometheus_shard
+
+            image_version = self.get_image_version()
+            if image_version is not None:
+                complete_config.metadata.labels[
+                    "paasta.yelp.com/image_version"
+                ] = image_version
+
+            complete_config.metadata.labels["paasta.yelp.com/remote_run"] = "true"
+            # DO NOT ADD LABELS AFTER THIS LINE
+            config_hash = get_config_hash(
+                self.sanitize_for_config_hash(complete_config),
+                force_bounce=self.get_force_bounce(),
+            )
+            complete_config.metadata.labels["yelp.com/paasta_config_sha"] = config_hash
+            complete_config.metadata.labels["paasta.yelp.com/config_sha"] = config_hash
+
+            complete_config.spec.template.metadata.labels[
+                "yelp.com/paasta_config_sha"
+            ] = config_hash
+            complete_config.spec.template.metadata.labels[
+                "paasta.yelp.com/config_sha"
+            ] = config_hash
```

Comment:
this stuff is very related to deployment bounces, so I'd say we don't need it in this case